### PR TITLE
[7.5.x] Cover XLS conversion by CDI test

### PR DESCRIPTION
**Pull request by jomarko**
Cherry pick of #728 
@manstis please have a look when you have a moment

### CODE REVIEW

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.kie.workbench.common.services.shared.preferences.ApplicationPreferences;
+import org.uberfire.backend.server.util.Paths;
+import org.uberfire.backend.vfs.Path;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DecisionTableXLSServiceImplCDITest extends CDITestSetup {
+
+    private ConversionResult conversionResult;
```
Comment (triceo) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java]: Any reason for having this as a member variable? Why not just return from `convertResource()` directly? 
Stateless code is better code.

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * Covers RHBA-48 - BigDecimal fields in table
+     */
+    @Test
+    public void testConvertFunctionAndBigDecimal() throws Exception {
+        convertResource("dtables/src/main/resources/guvnor/feature/dtables/FunctionAndBigDecimalTable.xls");
+        assertEquals(1, conversionResult.getMessages().size());
+        assertTrue(conversionResult.getMessages().get(0).getMessage().startsWith("Created Guided Decision Table 'FunctionAndBigDecimalTable (converted on"));
+        assertTrue(conversionResult.getMessages().get(0).getMessage().endsWith(").gdst'"));
```
Comment (triceo) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java]: final List<...> messages = conversionResult.getMessages();
    Assertions.assertThat(messages).hasSize(1);
    final String message = messages.get(0).getMessage();
    SoftAssertions.softly(softly -> {
        softly.assertThat(message).startsWith(...);
        softly.assertThat(message).endsWIth(...);
    });

Better, isn't it?

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    @Test
+    public void testConvertStaticFields() throws Exception {
+        convertResource("dtables/src/main/resources/guvnor/feature/dtables/StaticFieldsTable.xls");
+        assertEquals(2, conversionResult.getMessages().size());
+        assertEquals("Unable to convert value 'Message.GOODBYE' to NUMERIC_INTEGER. Cell (F11)",
+                     conversionResult.getMessages().get(0).getMessage());
+        assertTrue(conversionResult.getMessages().get(1).getMessage().startsWith("Created Guided Decision Table 'StaticFieldsTable (converted on"));
+        assertTrue(conversionResult.getMessages().get(1).getMessage().endsWith(").gdst'"));
```
Comment (triceo) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java]: Similar updates as above also possible here.

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/org/drools/workbench/screens/dtablexls/backend/server/dtables/src/main/java/guvnor/feature/dtables/Person.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        this.value = value;
+        this.name = name;
+        this.dummy = dummy;
+        this.sex = sex;
+        this.age = age;
+    }
+
+
+    
+    public java.lang.String getSex() {
```
Comment (triceo) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/org/drools/workbench/screens/dtablexls/backend/server/dtables/src/main/java/guvnor/feature/dtables/Person.java]: Any reason for the FQN here?

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java (hunk 1 above)
Comment (jomarko) [reply drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java]: Will do.

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java (hunk 2 above)
Comment (jomarko) [reply drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImplCDITest.java]: I Agree, will apply.

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/org/drools/workbench/screens/dtablexls/backend/server/dtables/src/main/java/guvnor/feature/dtables/Person.java (hunk 4 above)
Comment (jomarko) [reply drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/org/drools/workbench/screens/dtablexls/backend/server/dtables/src/main/java/guvnor/feature/dtables/Person.java]: Reason is that the tested project is copied directly from RHDM, however I am fine with updating this to the usual not FQN form.

**Pull request merged**